src/main/java/microservices/userServiceBasic/domain/UserService.java

```diff
@@ -0,0 +1,44 @@
+package microservices.userServiceBasic.domain;
+
+import microservices.userServiceBasic.domain.exception.UserNotFoundException;
+import microservices.userServiceBasic.domain.mapper.UserMapper;
+import microservices.userServiceBasic.domain.model.User;
+import microservices.userServiceBasic.domain.validation.EmailValidation;
+import microservices.userServiceBasic.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class UserService {
+
+	private final EmailValidation emailValidation;
+	private final UserRepository userRepository;
+	private final UserMapper userMapper;
+
+	@Autowired
+	public UserService(EmailValidation emailValidation, UserRepository userRepository, UserMapper userMapper) {
+
+		this.emailValidation = emailValidation;
+		this.userRepository = userRepository;
+		this.userMapper = userMapper;
+	}
+
+
+	public User saveUser(User user) {
+
+		if ( !emailValidation.isValidEmailAddress(user.getEmail()) ) {
+			throw emailValidation.emailValidationException;
```

Comment:
Not sure what you're trying to do here.. :D Take a look here: https://github.com/PauloFer1/pf-onboard/blob/master/src/main/java/com/pfernand/pfonboard/pfonboard/core/exception/InvalidEmailException.java
and where I call it here: https://github.com/PauloFer1/pf-onboard/blob/master/src/main/java/com/pfernand/pfonboard/pfonboard/core/validation/ValidateUserOnboard.java#L50